pkg/health/server/server.go

```diff
@@ -99,35 +105,55 @@ func (s *Server) DumpUptime() string {
 	return time.Since(s.startTime).String()
 }
 
-// getNodes fetches the latest set of nodes in the cluster from the Cilium
-// daemon, and updates the Server's 'nodes' map.
-func (s *Server) getNodes() (nodeMap, error) {
+// getNodes fetches the nodes added and removed from the last time the server
+// made a request to the daemon. If `allNodes` is set, it will return all nodes
+// the daemon is aware of.
+func (s *Server) getNodes(allNodes bool) (nodeMap, nodeMap, error) {
 	scopedLog := log
 	if s.CiliumURI != "" {
 		scopedLog = log.WithField("URI", s.CiliumURI)
 	}
-	scopedLog.Debug("Sending request for /healthz ...")
+	scopedLog.Debug("Sending request for /cluster/nodes ...")
 
-	resp, err := s.Daemon.GetHealthz(nil)
+	var clusterNodesParam *daemon.GetClusterNodesParams
+	if !allNodes {
+		clusterNodesParam = daemon.NewGetClusterNodesParams()
+		clusterNodesParam.SetClientID(&s.clientID)
+	}
+
+	resp, err := s.Daemon.GetClusterNodes(clusterNodesParam)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get agent health: %s", err)
+		return nil, nil, fmt.Errorf("unable to get agent health: %s", err)
+	}
+	log.Debug("Got cilium /cluster/nodes")
+
+	if resp == nil || resp.Payload == nil {
+		return nil, nil, fmt.Errorf("received nil health response")
 	}
-	log.Debug("Got cilium /healthz")
 
-	if resp == nil || resp.Payload == nil || resp.Payload.Cluster == nil {
-		return nil, fmt.Errorf("received nil health response")
+	if !allNodes {
+		s.clientID = resp.Payload.ClientID
 	}
 
-	if resp.Payload.Cluster.Self != "" {
+	if resp.Payload.Self != "" {
```

Comment:
Will `Self` always be populated, or only during the initial dump?

Reply:
Always.